assembl/tests/views_tests/test_graphql.py

```diff
@@ -751,6 +751,54 @@ def test_mutation_create_post_without_subject(graphql_request, thematic_and_ques
     }}}
 
 
+def test_mutation_create_post_on_column(graphql_request,
+                                        test_session,
+                                        idea_message_column_positive):
+    idea_id = to_global_id('Idea', idea_message_column_positive.idea_id)
+    classifier = idea_message_column_positive.message_classifier
+    res = schema.execute(u"""
+mutation myFirstMutation {
+    createPost(
+        ideaId:"%s",
+        subject:"Proposition 1",
+        body:"une proposition...",
+        messageClassifier:"%s"
+    ) {
+        post {
+            ... on Post {
+                subject
+                body
+                bodyEntries { localeCode value }
+                creator { name }
+                bodyMimeType
+                publicationState
+                messageClassifier
+            }
+        }
+    }
+}
+""" % (idea_id, classifier), context_value=graphql_request)
+
+    # Must remove the ICL before test completes in order to avoid db constraint
```

Comment:
I don't understand?

Reply:
The mutation in the test creates a post and an `idea_content_link`. When the test ends, pytest tries to run the finalizer of the fixtures, including an `idea` fixture. However, as there is an existing `idea_content_link` with a constraint of always having an `idea`, the fixture finalizer crashes. So, must remove ICL before test ends for smooth operation.

Reply:
When the post is removed, the idea_content_link row is automatically removed by a database cascade, no? You mean it is not possible to remove the post first and then the idea? Or maybe consider having a delete cascade on idea_content_link rows when the idea is deleted?

Reply:
Well, the test creates a post via graphql. So it's not a fixture to be automatically removed. Therefore, when the test ends, it's not removed to cascade to ICLs.